deepmd/pt/model/descriptor/dpa2.py

```diff
@@ -396,3 +399,32 @@ def forward(
         if self.concat_output_tebd:
             g1 = torch.cat([g1, g1_inp], dim=-1)
         return g1, rot_mat, g2, h2, sw
+
+    @classmethod
+    def update_sel(cls, global_jdata: dict, local_jdata: dict):
+        """Update the selection and perform neighbor statistics.
+
+        Parameters
+        ----------
+        global_jdata : dict
+            The global data, containing the training section
+        local_jdata : dict
+            The local data refer to the current class
+        """
+        local_jdata_cpy = local_jdata.copy()
+        update_sel = UpdateSel()
+        local_jdata_cpy = update_sel.update_one_sel(
+            global_jdata,
+            local_jdata_cpy,
+            True,
+            rcut_key="repinit_rcut",
+            sel_key="repinit_nsel",
```

Comment:
the interface of update_sel may be refactorized. left for a future PR.

Reply:
Perhaps the interface of dpa2 should be refactorized. For example,

```json
{
  "type": "dpa2",
  "repinit": {
    "rcut": 6,
    "rcut_smth": 0.5,
    "sel": "auto",
  },
  "repformer": {
    "rcut": 6,
    "rcut_smth": 0.5,
    "sel": "auto",
  }
}
```

Reply:
cc @iProzd

Reply:
the input output of update_sel are dicts, which is not a good design.